mono/metadata/marshal.c

```diff
@@ -6264,6 +6264,27 @@ mono_marshal_asany_impl (MonoObjectHandle o, MonoMarshalNative string_encoding,
 
 		return res;
 	}
+	case MONO_TYPE_SZARRAY: {
+		MonoClass *klass = t->data.klass;
+		MonoClass *eklass = m_class_get_element_class (klass);
+		if (m_class_get_rank (klass) > 1)
+			break;
+
+		if (mono_class_is_auto_layout (eklass))
+			break;
+		if (m_class_is_valuetype (eklass) && (mono_class_is_explicit_layout (eklass) || m_class_is_blittable (eklass) || m_class_is_enumtype (eklass)))	{
+			//FIXME: Pin array - if we can't pin, we should probably do a copy here instead of this
```

Comment:
@vargaz Are we okay not pinning here? Can we assume the array will be pinned because a reference will be on the stack of a running thread?